api/views/uqj_generic.py

```diff
@@ -0,0 +1,27 @@
+from rest_framework import viewsets
+from course.models.models import UserQuestionJunction
+from api.serializers import UQJSerializer
+
+from rest_framework.decorators import action
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+
+
+class UQJGenericViewSet(viewsets.GenericViewSet):
+    """
+    Query Parameters
+    + Standard ordering is applied on the field 'last_viewed'
+    """
+    serializer_class = UQJSerializer
+
+    @action(detail=False, methods=['post'], url_path='switch-favorite')
+    def switch_favorite(self, request, pk=None):
```

Comment:
switch favorite is a misleading name. `update_is_favorite` would be a better one.